lib/charms/layer/containerd.py

```diff
@@ -19,6 +19,15 @@ class ResourceFailure(Exception):
     pass
 
 
+def can_mount_cgroup2() -> bool:
+    """Determine if it's possible to mount cgroup2 type filesystems."""
+    try:
+        stdout = check_output(["mount", "-t", "cgroup2"], text=True)
```

Comment:
This would be true for focal too. The main difference between focal and jammy is whether cgroup v1 is mounted or not.

So yes, we need to use io.containerd.runc.v2 for jammy, but not sure we want to switch to io.containerd.runc.v2 for focal in a rush.


[focal]
```
$ mount | grep ^cgroup
cgroup2 on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset,clone_children)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
cgroup on /sys/fs/cgroup/rdma type cgroup (rw,nosuid,nodev,noexec,relatime,rdma)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
```


[jammy]
```
$ mount | grep ^cgroup
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime)
```

Reply:
i was just reproducing for consistency what was implemented for [microk8s](https://github.com/charmed-kubernetes/charm-containerd/pull/89)  Is there a good reason to differentiate?

Reply:
also [the recommendation](https://github.com/containerd/containerd/discussions/6446) from containerd is to drop `io.containerd.runc.v1` as it's been deprecated

Reply:
I don't see any problem with focal getting updated too. We will be testing this change on both jammy and focal before we release.

Reply:
since this charm only now supports focal and jammy -- should we just adjust the templates to use `v2` and end the nonsense about checking if we can mount cgroup2 types?